sdk/identity/identity/migration-v1-v2.md

```diff
@@ -0,0 +1,159 @@
+# Migrate from v1 to v2 of @azure/identity
+
+[v2changelog]: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/CHANGELOG.md#200-2021-10-12
+[plugins]: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/CHANGELOG.md#200-2021-10-12#plugin-api
+[npm-keytar]: https://www.npmjs.com/package/keytar
+[npm-vscode]: https://www.npmjs.com/package/@azure/identity-vscode
+
+The `@azure/identity` v2 package includes the best parts of v1, plus several improvements. This document outlines the steps needed to migrate from v1 to v2. For the full set of changes and bug fixes, refer to the [changelog][v2changelog].
+
+## Table of contents
+
+- [Update packages](#update-packages)
+  - [Update Identity package](#update-identity-package)
+  - [Update dependencies for Visual Studio Code authentication](#update-visual-studio-code-authentication)
+- [Update error handling](#update-error-handling)
+  - [Update authentication error handling](#update-authentication-error-handling)
+  - [Update error handling code for getToken calls](#update-error-handling-code-for-gettoken-calls)
+  - [Update ClientCertificateCredential error handling](#update-clientcertificatecredential-error-handling)
+- [Update app registration for browser authentication](#update-app-registration-for-browser-authentication)
+- [Troubleshooting](#troubleshooting)
+- [Provide feedback](#provide-feedback)
+- [Contributing](#contributing)
+
+## Update packages
+
+### Update Identity package
+
+Run the following command to update to Identity v2:
+
+```bash
+npm install --save @azure/identity@^2.0.0
+```
+
+The *package.json* file key-value pair now resembles the following:
+
+```diff
+  "dependencies": {
+-    "@azure/identity": "^1.5.2",
++    "@azure/identity": "^2.0.0",
+  }
+```
+
+### Update dependencies for Visual Studio Code authentication
+
+If you use the `VisualStudioCodeCredential`, you likely have [keytar][npm-keytar] in your dependencies. `keytar` is a popular native machine-code (NMC) dependency in the Node.js ecosystem. NMC dependencies were removed from `@azure/identity` and put into individual packages. To use the `VisualStudioCodeCredential` with Identity v2, complete the following steps:
```

Comment:
NMC -> Is this a popular abbreviation or something we made up? If it is not a well known abbreviation, then I would recommend not using it.

Reply:
I think it’s a .NET thing.